pkg/controllers/nodeclaim/podevents/suite_test.go

```diff
@@ -0,0 +1,169 @@
+	RunSpecs(t, "Disruption")
+}
+
+var _ = BeforeSuite(func() {
+	fakeClock = clock.NewFakeClock(time.Now())
+	env = test.NewEnvironment(test.WithCRDs(apis.CRDs...), test.WithCRDs(v1alpha1.CRDs...), test.WithFieldIndexers(func(c cache.Cache) error {
+		return c.IndexField(ctx, &v1.NodeClaim{}, "status.providerID", func(obj client.Object) []string {
+			return []string{obj.(*v1.NodeClaim).Status.ProviderID}
+		})
+	}))
+	ctx = options.ToContext(ctx, test.Options())
+	cp = fake.NewCloudProvider()
+	podEventsController = podevents.NewController(fakeClock, env.Client)
+})
+
+var _ = AfterSuite(func() {
+	Expect(env.Stop()).To(Succeed(), "Failed to stop environment")
+})
+
+var _ = BeforeEach(func() {
+	ctx = options.ToContext(ctx, test.Options())
+	fakeClock.SetTime(time.Now())
+})
+
+var _ = AfterEach(func() {
+	cp.Reset()
+	ExpectCleanedUp(ctx, env.Client)
+})
+var _ = Describe("PodEvents", func() {
+	var nodePool *v1.NodePool
+	var nodeClaim *v1.NodeClaim
+	var node *corev1.Node
+	var pod *corev1.Pod
+
+	BeforeEach(func() {
+		nodePool = test.NodePool()
+		nodeClaim, node = test.NodeClaimAndNode(v1.NodeClaim{
+			ObjectMeta: metav1.ObjectMeta{
+				Labels: map[string]string{
+					v1.NodePoolLabelKey:            nodePool.Name,
+					corev1.LabelInstanceTypeStable: "default-instance-type", // need the instance type for the cluster state update
+				},
+			},
+			Status: v1.NodeClaimStatus{
+				ProviderID: test.RandomProviderID(),
+			},
+		})
+		pod = test.Pod(test.PodOptions{
+			NodeName: node.Name,
+		})
+	})
+	It("should set the nodeclaim lastPodEvent", func() {
+		ExpectApplied(ctx, env.Client, nodePool, nodeClaim, node, pod)
+		fakeClock.SetTime(pod.CreationTimestamp.Time)
+		ExpectObjectReconciled(ctx, env.Client, podEventsController, pod)
+
+		nodeClaim = ExpectExists(ctx, env.Client, nodeClaim)
+		Expect(nodeClaim.Status.LastPodEventTime.Time).To(BeEquivalentTo(pod.CreationTimestamp.Time))
+	})
+	It("should not set the nodeclaim lastPodEvent for a node that Karpenter doesn't own", func() {
+		delete(node.Labels, v1.NodePoolLabelKey)
+		ExpectApplied(ctx, env.Client, nodePool, node, pod)
+		fakeClock.SetTime(pod.CreationTimestamp.Time)
+		ExpectObjectReconciled(ctx, env.Client, podEventsController, pod)
+
+		Expect(nodeClaim.Status.LastPodEventTime.Time).To(BeZero())
+	})
+	It("should not set the nodeclaim lastPodEvent when the node does not exist", func() {
+		ExpectApplied(ctx, env.Client, nodePool, nodeClaim, pod)
+		fakeClock.SetTime(pod.CreationTimestamp.Time)
+		_ = ExpectObjectReconcileFailed(ctx, env.Client, podEventsController, pod)
+
+		nodeClaim = ExpectExists(ctx, env.Client, nodeClaim)
+		Expect(nodeClaim.Status.LastPodEventTime.Time).To(BeZero())
+	})
+	It("should not set the nodeclaim lastPodEvent when the nodeclaim does not exist", func() {
+		ExpectApplied(ctx, env.Client, nodePool, node, pod)
+		fakeClock.SetTime(pod.CreationTimestamp.Time)
+		_ = ExpectObjectReconcileFailed(ctx, env.Client, podEventsController, pod)
```

Comment:
Seems like expecting the reconciliation to fail should be used extremely sparingly since this means that we are returning errors and we generally want to avoid returning errors back to the user